src/main/java/org/generationcp/middleware/service/impl/user/UserServiceImpl.java

```diff
@@ -135,54 +140,52 @@ public Integer createUser(final UserDto userDto) {
 		// user.status = 0 - Unassigned
 		// user.type = 0 - Default user type (not used)
 
-		final Integer currentDate = Util.getCurrentDateAsIntegerValue();
-		final Person person = this.createPersonFromDto(userDto, new Person());
-
-		final WorkbenchUser user = new WorkbenchUser();
-		user.setPerson(person);
-		user.setName(userDto.getUsername());
-		user.setPassword(userDto.getPassword());
-		user.setAccess(0);
-		user.setAssignDate(currentDate);
-		user.setCloseDate(currentDate);
-		user.setInstalid(0);
-		user.setStatus(userDto.getStatus());
-		user.setType(0);
-
-		// Add user roles to the particular user
-		final List<UserRole> userRoles = new ArrayList<>();
-		if (userDto.getUserRoles() != null) {
-			for (final UserRoleDto userRoleDto : userDto.getUserRoles()) {
-				userRoles.add(this.buildNewUserRole(user, userRoleDto));
-			}
-		}
-		user.setRoles(userRoles);
+		try {
+			final Integer currentDate = Util.getCurrentDateAsIntegerValue();
+			final Person person = this.createPersonFromDto(userDto, new Person());
 
-		final Set<CropType> crops = new HashSet<>();
-		for (final CropDto crop : userDto.getCrops()) {
-			final CropType cropType = new CropType(crop.getCropName());
-			crops.add(cropType);
-		}
-		person.setCrops(crops);
+			final WorkbenchUser user = new WorkbenchUser();
+			user.setPerson(person);
+			user.setName(userDto.getUsername());
+			user.setPassword(userDto.getPassword());
+			user.setAccess(0);
+			user.setAssignDate(currentDate);
+			user.setCloseDate(currentDate);
+			user.setInstalid(0);
+			user.setStatus(userDto.getStatus());
+			user.setType(0);
 
-		try {
+			// Add user roles to the particular user
+			final List<UserRole> userRoles = new ArrayList<>();
+			if (userDto.getUserRoles() != null) {
+				for (final UserRoleDto userRoleDto : userDto.getUserRoles()) {
+					userRoles.add(this.buildNewUserRole(user, userRoleDto));
+				}
+			}
+			user.setRoles(userRoles);
+
+			final Set<CropType> crops = new HashSet<>();
+			for (final CropDto crop : userDto.getCrops()) {
+				final CropType cropType = new CropType(crop.getCropName());
+				crops.add(cropType);
+			}
+			person.setCrops(crops);
 
 			final WorkbenchUser recordSaved = this.workbenchDaoFactory.getWorkbenchUserDAO().saveOrUpdate(user);
 			idUserSaved = recordSaved.getUserid();
```

Comment:
@lik-83 we dont need this variable right? we can return recordSaved.getUserid() at the end of the try block